addons/mrp/models/stock_rule.py

```diff
@@ -215,6 +215,21 @@ def run(self, procurements, raise_user_error=True):
                     quant_uom = bom_line.product_id.uom_id
                     # recreate dict of values since each child has its own bom_line_id
                     values = dict(procurement.values, bom_line_id=bom_line.id)
+                    # Remove from product_description_variants in procurement values
+                    # any description that does not apply to the actual component
+                    if values.get("product_description_variants"):
+                        lang = self._get_product_description_variants_lang(values)
+                        kit_product_description_variants_list = values.get("product_description_variants").lstrip("\n").split("\n")
+                        valid_product_ptavs = bom_line.product_id.product_tmpl_id.valid_product_template_attribute_line_ids.product_template_value_ids
+                        product_description_variant_list = [
+                            kpdvl
+                            for kpdvl in kit_product_description_variants_list
+                            if kpdvl.split(":")[0] in [
+                                vpptavs.split(":")[0]
+                                for vpptavs in valid_product_ptavs.with_context(lang=lang).mapped("display_name")
+                            ]
+                        ]
+                        values["product_description_variants"] = "\n\n%s" % "\n".join(product_description_variant_list)
```

Comment:
I'm wondering if there is not another place for that.
Couldn't those useless attribute values be filtered when this description is created instead?

Reply:
The attribute values are not useless and are quite meaningful if we don't want to manage a variant for each value. 

We cannot remove their creation because it makes sense for the kit product, but it doesn't necessarily when the kit is "exploded" into its parts, and this happens in the mrp module on the code I did modify here.